### DISCUSSION TITLE
Question: How to force to reload data for sever side Datagrid

### DISCUSSION THREAD

**Discussion by mschnecke** [blazority/support#118] [Q&A] [answered]
Sample here: https://github.com/mschnecke/demo-blazority

Detail:
https://github.com/mschnecke/demo-blazority/blob/main/src/Demo.Blazor.Clarity/Client/Pages/Index.razor.cs
after creating a new user in the backend I would like to refresh the data in the frontend.

await Datagrid.RefreshViewItems(true); <= seems not to work

**Comment by karan-kang** [comment]
This should be working, you can see the "Reload" button implementation on Blazority site:
https://blazority.com/datagrid/server-side

```

<Button @onclick="() => ReloadItems()" Small="true">Reload</Button>

public async Task ReloadItems()
    {
        if (Datagrid != null)
        {
            // The "true" value ensured we always refresh items from server regardless of whether there are changes
            // to filtering, sorting, or pagination
            await Task.Yield();
            await Datagrid.RefreshViewItems(true);
        }
    }
```

**Comment by karan-kang** [comment]
I will try this with your github example as well, thanks for creating the sample project.

**Comment by karan-kang** [comment, marked as answer]
@mschnecke In your example code, you forgot to set the @ref object to Datagrid in the Index.razor template file.

![image](https://user-images.githubusercontent.com/1234394/197420300-1a9e88e5-5d6f-43e2-9eb1-f54d6a639b0c.png)

In the Index.razor.cs file, set the Datagrid property as following instead of using **new()** keyword
![image](https://user-images.githubusercontent.com/1234394/197420332-eca3634f-dbf6-47c2-bf84-d325557b838b.png)

Once this change was made, the Reload method worked properly since now it is pointing to correct Datagrid reference. I see new items getting added to the server and reload of data items:
![image](https://user-images.githubusercontent.com/1234394/197420395-7b674602-0c38-4b0a-84d4-3fd63fcac418.png)


please feel free to reopen if you need additional help

**Comment by mschnecke** [comment]
It works. Thank you. Btw. the Datagrid documentation for server side has no @ref attribute.
